dipy/reconst/dti.py

```diff
@@ -769,16 +825,25 @@ def fit(self, data, mask=None):
             A boolean array used to mark the coordinates in the data that
             should be analyzed that has the shape data.shape[:-1]
 
+        adjacency : float, optional
+            Calculate voxel adjacency accounting for mask, using this
+            value as cutoff distance.
         """
+
         S0_params = None
 
+        img_shape = data.shape[:-1]
         if mask is not None:
             # Check for valid shape of the mask
-            if mask.shape != data.shape[:-1]:
+            if mask.shape != img_shape:
                 raise ValueError("Mask is not the same shape as data.")
             mask = np.array(mask, dtype=bool, copy=False)
         data_in_mask = np.reshape(data[mask], (-1, data.shape[-1]))
 
+        if adjacency > 0:
```

Comment:
strange that this work. I did not know that you could do ` False > 0` or `True >0`. Interesting.  I need to get some info for that.

For now, I would recommend to do `if adjacency and adjacency > 0`

Reply:
How about initializing with zero then, therefore we only calculate adjacency if a value above zero is given?

Reply:
I think a test should be added of the adjacency calculation in the context of the model.